src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultDocumentVersionCache.cs

```diff
@@ -15,7 +18,20 @@ internal class DefaultDocumentVersionCache : DocumentVersionCache
         // Internal for testing
         internal readonly Dictionary<string, List<DocumentEntry>> DocumentLookup;
         private readonly ProjectSnapshotManagerDispatcher _projectSnapshotManagerDispatcher;
-        private ProjectSnapshotManagerBase _projectSnapshotManager;
+        private ProjectSnapshotManagerBase? _projectSnapshotManager;
+
+        private ProjectSnapshotManagerBase ProjectSnapshotManager
+        {
+            get
+            {
+                if (_projectSnapshotManager is null)
+                {
+                    throw new InvalidOperationException("ProjectSnapshotManager accessed before Initialized was called.");
```

Comment:
Is this helpful? This is just hiding a potential null access isn't it?

Reply:
I did it because it hides the potential null-access, but I actually do like this pattern because this throw ever being hit indicates a serious and specific issue (Server failing to initialize everything before accepting requests.